File: Sacred-Ecommerce-Microservice/carts/src/main/java/com/sacred/carts/audit/auditAwareImpl.java
package com.sacred.carts.audit;


import org.springframework.data.domain.AuditorAware;
import org.springframework.stereotype.Component;

import java.util.Optional;

@Component(value = "cartAuditAware")
public class auditAwareImpl implements AuditorAware<String> {
    @Override
    public Optional<String> getCurrentAuditor() {
        return Optional.of("CARTS_MS");
    }
}

File: Sacred-Ecommerce-Microservice/orderManagement/src/main/kotlin/com/sacred/ordermanagement/audit/AuditAwareImpl.kt
package com.sacred.ordermanagement.audit

import org.springframework.data.domain.AuditorAware
import org.springframework.stereotype.Component
import java.util.Optional

@Component("auditAwareImpl")
class AuditAwareImpl: AuditorAware<String>{
    override fun getCurrentAuditor(): Optional<String> {
        return Optional.of("ORDER_MGMT_MS")
    }
}